### PR TITLE
[4.0] Add prepared statements for mod_whosonline

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the module in all ways you can think of.

* create a frontend module
* set the filter by group
* log into frontend with a non admin user
* You should see all logged in users with the access level your user have access to.

### Expected result
Nothing changed.

### CODE REVIEW

File: modules/mod_whosonline/Helper/WhosonlineHelper.php [hunk 1]
```diff
@@ -111,8 +111,8 @@ public static function getOnlineUserNames($params)
 
 			$query->join('LEFT', '#__user_usergroup_map AS m ON m.user_id = a.userid')
```
Comment (Quy) [inline modules/mod_whosonline/Helper/WhosonlineHelper.php]: Apply `$db->quoteName`

File: modules/mod_whosonline/Helper/WhosonlineHelper.php [hunk 2]
```diff
@@ -93,7 +93,7 @@ public static function getOnlineUserNames($params)
 		$db    = Factory::getDbo();
 		$query = $db->getQuery(true)
 			->select($db->quoteName(array('a.username', 'a.userid', 'a.client_id')))
-			->from('#__session AS a')
+			->from($db->quoteName('#__session', 'a'))
 			->where($db->quoteName('a.userid') . ' != 0')
 			->where($db->quoteName('a.client_id') . ' ' . $whereCondition)
 			->group($db->quoteName(array('a.username', 'a.userid', 'a.client_id')));
```
Comment (Quy) [inline modules/mod_whosonline/Helper/WhosonlineHelper.php]:
```suggestion
			->group($db->quoteName(['a.username', 'a.userid', 'a.client_id']));
```

File: modules/mod_whosonline/Helper/WhosonlineHelper.php [hunk 3]
```diff
@@ -109,10 +109,10 @@ public static function getOnlineUserNames($params)
 				return array();
 			}
 
-			$query->join('LEFT', '#__user_usergroup_map AS m ON m.user_id = a.userid')
-				->join('LEFT', '#__usergroups AS ug ON ug.id = m.group_id')
-				->where('ug.id in (' . implode(',', $groups) . ')')
-				->where('ug.id <> 1');
+			$query->leftJoin($db->quoteName('#__user_usergroup_map', 'm') . ' ON ' . $db->quoteName('m.user_id') . ' = ' . $db->quoteName('a.userid'))
```
Comment (Quy) [inline modules/mod_whosonline/Helper/WhosonlineHelper.php]:
```suggestion
			$query->leftJoin($db->quoteName('#__user_usergroup_map', 'm'),  $db->quoteName('m.user_id') . ' = ' . $db->quoteName('a.userid'))
```

File: modules/mod_whosonline/Helper/WhosonlineHelper.php [hunk 4]
```diff
@@ -109,10 +109,10 @@ public static function getOnlineUserNames($params)
...
+				->leftJoin($db->quoteName('#__usergroups', 'ug') . ' ON ' . $db->quoteName('ug.id') . ' = ' . $db->quoteName('m.group_id'))
```
Comment (Quy) [inline modules/mod_whosonline/Helper/WhosonlineHelper.php]:
```suggestion
				->leftJoin($db->quoteName('#__usergroups', 'ug'), $db->quoteName('ug.id') . ' = ' . $db->quoteName('m.group_id'))
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 09114fd67dc8bcf0f6f38ab5935187f221642f90<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25044">issues.joomla.org/tracker/joomla-cms/25044</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 09114fd67dc8bcf0f6f38ab5935187f221642f90<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25044">issues.joomla.org/tracker/joomla-cms/25044</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25044">issues.joomla.org/tracker/joomla-cms/25044</a>.</sub>

**Comment by roland-d** [comment]
Thank you.

**Pull request merged**